nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -2007,6 +2057,265 @@ impl DataStore {
                 }
             })
     }
+
+    /// Replace a read-only snapshot in a Volume with a new region
+    ///
```

Comment:
I wonder if using "snapshot" here is not strictly true.
The volume has no concept of a snapshot vs. a read_only_region.  All
it really knows are the targets.

Reply:
da91ba306 fixes this